lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -798,6 +769,168 @@ def accessible_objects(user_or_attribute, operation, options = {})
 
     private
 
+    # Given a policy element class and a set of conditions, returns an
+    # ActiveRecord_Relation with those conditions applied
+    def build_active_record_relation(pe_class:, conditions:, ignore_case:)
+      if ignore_case
+        # If any condition is case-insensitive, the nodes need to be built
+        # individually with Arel.
+        build_ignore_case_relation(
+          pe_class: pe_class,
+          conditions: conditions,
+          ignore_case: ignore_case
+        )
+      else
+        # If all conditions are case-sensitive, a direct where call can be used.
+        pe_class.where(conditions.to_h)
+      end
+    end
+
+    def build_ignore_case_relation(pe_class:, conditions:, ignore_case:)
+      # Conditions hash to array so it can be reduced
+      condition_array = conditions.to_a
+      # Initialize an "empty" relation
+      starting_relation = pe_class.where(nil)
+
+      # Reduce the conditions array to a single relation by building Arel nodes
+      condition_array.reduce(starting_relation) do |relation, condition_pair|
+        key, value = condition_pair
+
+        arel_node = if ignore_case_applies?(ignore_case, key)
+                      build_arel_insensitive(pe_class: pe_class, key: key, value: value)
+                    else
+                      build_arel_sensitive(pe_class: pe_class, key: key, value: value)
+                    end
+
+        relation.where(arel_node)
+      end
+    end
+
+    # Build Arel nodes using case-insensitive matching
+    def build_arel_insensitive(pe_class:, key:, value:)
+      if value.is_a?(Array)
+        pe_class.arel_table[key].matches_any(value)
+      else
+        pe_class.arel_table[key].matches(value)
+      end
+    end
+
+    # Build Arel nodes using case-sensitive equality checking
+    def build_arel_sensitive(pe_class:, key:, value:)
+      # Arel blows up with empty array passed to eq_any
+      # See: https://github.com/rails/arel/issues/368
+      if value.is_a?(Array) && value.present?
+        pe_class.arel_table[key].eq_any(value)
+      elsif value.is_a?(Array) && value.empty?
+        ::Arel::Nodes::SqlLiteral.new("(NULL)")
+      else
+        pe_class.arel_table[key].eq(value)
+      end
+    end
+
+    # Iterate over the input scope and return elements that match the extra
+    # attribute conditions
+    def filter_by_extra_attributes(scope:, extra_attribute_conditions:, pe_class:, ignore_case:)
+      if extra_attribute_conditions.empty?
+        scope
+      else
+        ids = scope.reduce([]) do |memo, policy_element|
+          pe_within_scope = extra_attribute_conditions.all? do |key, value|
+            pe_matches_extra_attributes?(policy_element, key, value, ignore_case)
+          end
+
+          memo.push(policy_element.id) if pe_within_scope
```

Comment:
We could use a ternary operator if that's your kind of thing. I'm also fine with this, though.

Reply:
I'm not sure what you mean - I don't really see how a ternary could be used here. Mind sharing more detail?

Reply:
Like
```ruby
pe_within_scope ? memo.push(policy_element.id) : memo
```